psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,795 @@
+        The molecule to displace
+    freq_irrep_only : int
+        The Cotton ordered irrep to get frequencies for. Choose -1 for all
+        irreps.
+    mode : {"1_0", "2_0", "2_1"}
+         The first number specifies the derivative level determined from
+         displacements, and the second number is the level determined at.
+    initialize_string : function
+         A function that returns the string to print to show the caller was entered.
+         The string is both caller-specific and dependent on values determined
+         in this function.
+    verbose : int
+         Set to 0 to silence extra print information, regardless of the print level.
+         Used so the information is printed only during geometry generation, and not
+         during the derivative computation as well.
+
+    :returns: *dict*
+        Miscellaneous information required by callers.
+    """
+
+    core.print_out("\n         ----------------------------------------------------------\n")
+    core.print_out("                                   FINDIF\n")
+    core.print_out("                     R. A. King and Jonathon Misiewicz\n")
+    core.print_out("         ---------------------------------------------------------\n\n")
+
+    print_lvl = core.get_option("FINDIF", "PRINT")
+    num_pts = core.get_option("FINDIF", "POINTS")
+    disp_size = core.get_option("FINDIF", "DISP_SIZE")
+
+    data = {"print_lvl": print_lvl, "num_pts": num_pts, "disp_size": disp_size}
+
+    if print_lvl:
+        core.print_out(initialize_string(data))
+
+    # Get settings for CdSalcList, then get the CdSalcList.
+    method_allowed_irreps = 0x1 if mode == "1_0" else 0xFF
+    t_project = not core.get_global_option("EXTERN") and (not core.get_global_option("PERTURB_H"))
+    # core.get_option returns an int, but CdSalcList expect a bool, so re-cast
+    r_project = t_project and bool(core.get_option("FINDIF", "FD_PROJECT"))
+    salc_list = core.CdSalcList(mol, method_allowed_irreps, t_project, r_project)
+
+    n_atom = mol.natom()
+    n_irrep = salc_list.nirrep()
+    n_salc = salc_list.ncd()
+
+    if print_lvl and verbose:
+        core.print_out("    Number of atoms is {:d}.\n".format(n_atom))
+        if method_allowed_irreps != 0x1:
+            core.print_out("    Number of irreps is {:d}.\n".format(n_irrep))
+        core.print_out("    Number of {!s}SALCs is {:d}.\n".format("" if method_allowed_irreps != 0x1 else "symmetric ",
+                                                                 n_salc))
+        core.print_out("    Translations projected? {:d}. Rotations projected? {:d}.\n".format(t_project, r_project))
+
+    # TODO: Replace with a generator from a stencil to a set of points.
+    pts_dict = {
+        3: {
+            "sym": ((-1, ), (1, )),
+            "asym": ((-1, ), ),
+            "off": ((1, 1), (-1, -1))
+        },
+        5: {
+            "sym": ((-2, ), (-1, ), (1, ), (2, )),
+            "asym": ((-2, ), (-1, )),
```

Comment:
Any particular reason to choose neg rather than pos for the single-sided step?

Are "sym" and "asym" good labels, considering that symmetric modes have asymmetric potentials and asymmetric modes have symmetric potentials?

Reply:
90% sure that choosing the negative step was the C!FINDIF convention. I don't care either way.
`sym_irr` and `asym_irr` would be better, now that you mention it. I'll hold off on making any changes until you say you're done with the review. No need to clog the commit history.

Reply:
Ok on the negative step. Just curious.

I'll throw "oneside" and "twoside" into the ring as possible labels.

I confess to reviewing bits at a time. Push at convenience.

Reply:
@psi-rking oh, I definitely agree once you throw "stretch" in there, it's clear that the subject of the "(a)sym" is mode. It just struck me that in describing a stencil, there's room for confusion.